ocaml/testsuite/tests/typing-layouts-err-msg/value.ml

```diff
@@ -377,7 +396,7 @@ Error: This pattern matches values of type t_float64
        The layout of t_float64 is float64, because
          of the definition of t_float64 at line 5, characters 0-24.
        But the layout of t_float64 must be a sublayout of value, because
-         it's a term-level argument to a class constructor.
+         it's the type of a term-level argument to a class constructor.
```

Comment:
This is still not yet fixed.

Reply:
In a [previous comment](https://github.com/ocaml-flambda/flambda-backend/pull/2091#discussion_r1431890173):

> But this one really is a term-level argument. I think the problem is that Class_argument is used in too many places. Some are term-level, and some are type-level.

For the test case in question here:

```ocaml
class foo (x : t_float64) =
  object end;;
```

Should `x` be considered as a term-level argument?

Reply:
Yes -- `x` is very much a term level argument. Type arguments go before the class name (and in brackets); term arguments go afterwards.

Reply:
Then is the error message `it's the type of a term-level argument to a class constructor` here correct since it's referring to the type of `x`?

The message for type arguments has been updated so that in examples like the one below, a different message is displayed:

```ocaml
class ['a] c12_4 = object
  method x : 'a t_float64_id -> 'a t_float64_id = assert false
end;;
[%%expect{|
Line 2, characters 13-15:
2 |   method x : 'a t_float64_id -> 'a t_float64_id = assert false
                 ^^
Error: This type ('a : float64) should be an instance of type ('a0 : value)
       The layout of 'a is value, because
         it's a type argument to a class constructor.
       But the layout of 'a must overlap with float64, because
         of the definition of t_float64_id at line 2, characters 0-37.
|}];;
```

Reply:
Maybe I just read too quickly previously? Yes this looks fine.